core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -800,6 +801,11 @@ class ReplicaManager(val config: KafkaConfig,
       }
 
       logManager.truncateTo(partitionsToMakeFollower.map(partition => (new TopicAndPartition(partition), partition.getOrCreateReplica().highWatermark.messageOffset)).toMap)
+      partitionsToMakeFollower.foreach { partition =>
+        val topicPartitionOperationKey = new TopicPartitionOperationKey(partition.topic, partition.partitionId)
+        tryCompleteDelayedProduce(topicPartitionOperationKey)
```

Comment:
I was wondering about this as well, but I think we do this right? i.e., it makes a call to `Partition.checkEnoughReplicasReachOffset` which returns `(false, Errors.NOT_LEADER_FOR_PARTITION.code)` if the leader is non-local